utils/mirror_vfd/mirror_writer.c

```diff
@@ -763,7 +763,7 @@ do_write(struct mirror_session *session, const unsigned char *xmit_buf)
 
     /* Allocate the buffer once -- re-use between loops.
      */
-    buf = (char *)HDmalloc(sizeof(char) * H5FD_MIRROR_DATA_BUFFER_MAX);
+    buf = (char *)HDmalloc(sizeof(char) * (unsigned long)H5FD_MIRROR_DATA_BUFFER_MAX);
```

Comment:
The casts in mirror_writer.c line 766 and line 871 are needed to eliminate warnings that a floating point number is implicitly converted to size_t (unsigned long).  While the cast may eliminate the warning, is it better to #define H5FD_MIRROR_DATA_BUFFER_MAX to H5_GB in H5FDmirror_priv.h (from H5private.h line 446: #define H5_GB (1024.0F * 1024.0F * 1024.0F)) than to #define H5FD_MIRROR_DATA_BUFFER_MAX (1024 * 1024 * 1024), casting it there if necessary?  Just curious.

Reply:
@lrknox, I have difficulty in understanding your comment. 

1. Do you mean line 791 instead of line 871?
2. Do you mean `#define H5FD_MIRROR_DATA_BUFFER_MAX (1024 * 1024 * 1024)`  is better than the current `#define 5FD_MIRROR_DATA_BUFFER_MAX H5GB` in H5FDmirror_priv.h to remove casting?
3. What do you mean "there" in "casting it there if necessary"?

Reply:
1. yes, 791 is the line I meant.

2. Not necessarily better, but no casts would be required where it is used.  And it is unlikely that 1 GB would be changed to some value other than 1024 * 1024 * 1024.

3. H5FD_MIRROR_DATA_BUFFER_MAX is used in 2 places, lines 766 and 791, both of which require casts to avoid the warning.  By "there" I meant in the #define in src/H5FDmirror_priv.h line 34.  If there is a need to use H5_GB to define H5FD_MIRROR_DATA_BUFFER_MAX, can src/H5FDmirror_priv.h line 34 be #define H5FD_MIRROR_DATA_BUFFER_MAX (unsigned long)H5_GB /* 1 Gigabyte */?  Or if it isn't necessary to define H5FD_MIRROR_DATA_BUFFER_MAX in terms of H5_GB, then perhaps #define H5FD_MIRROR_DATA_BUFFER_MAX (1024 * 1024 * 1024).

Searching the code for other uses of H5_GB, there are 3:  1 in src/H5Dmpio.c in an expression that is cast to hsize_t for the size of a dataset, and 2 in src/H5timer.c that are cast to double in calculating bandwidth.  It appears from the comment in H5private.h that H5_GB was defined for this purpose and later used in the other files because the value was correct, if not the type.

My comment was intended more as a question than as a suggestion.

Reply:
`#define H5FD_MIRROR_DATA_BUFFER_MAX (1024 * 1024 * 1024)` looks like the right thing to do.